From 9d5d830e89306520cb463e19bb83b2f791729de0 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Sun, 28 Aug 2016 03:16:52 +0200
Subject: [PATCH 0462/1345] net: mvpp2x: silence complaints on probe deferral

commit  480a17a29e1d19abcbd728672859123c7465b7ec from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

On some platforms, the clocks might be registered by a platform
driver. When this is the case, the clock platform driver may very well
be probed after mvpp2 driver, in which case the first probe() invocation
of mvpp2 driver will receive -EPROBE_DEFER as the return value of
devm_clk_get().

The current code handles that as a normal error, and prints error message.
This patch silence the error message, in case of probe deferral.

This patch is part of patch-set which adds
mainline mvebu clock gating support [14/21].

Change-Id: Iea94b1e7d542ed50f6d66c88be79f7c5a6351047
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32399
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 56cd37e..8e20271 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -4586,7 +4586,8 @@ static int mv_pp2x_probe(struct platform_device *pdev)
 
 	err = mv_pp2x_platform_data_get(pdev, priv, &cell_index, &port_count);
 	if (err) {
-		dev_err(&pdev->dev, "mvpp2: platform_data get failed\n");
+		if (err != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "mvpp2: platform_data get failed\n");
 		goto err_clk;
 	}
 	priv->pp2_version = priv->pp2xdata->pp2x_ver;
-- 
1.7.9.5

